### PR TITLE
Validate interface name length in CLI

### PR DESCRIPTION
**Pull request by arfeigin**
This PR is accompanied by https://github.com/sonic-net/sonic-swss-common/pull/889 and https://github.com/sonic-net/sonic-buildimage/pull/20108

#### What I did
Validate interface name length does not exceed the limitation of IFNAMSIZ to align to kernel restrictions.

#### How I did it
Add validation checks in the relevant config functions for the following interface types:

vxlan
vlan
vrf
loopback
subinterface
portchannel

#### How to verify it
UT tests added

### CODE REVIEW

**Comment by arfeigin** [comment]
Checkers fail as https://github.com/sonic-net/sonic-swss-common/pull/889 should be merged first so it will compile.

**Comment by arfeigin** [comment]
@prsunny Can you please help reviewing this and the attached PRs?

**Review by nazariig — APPROVED** [review]

File: config/vxlan.py [hunk 1]
```diff
@@ -3,8 +3,10 @@
 
 from jsonpatch import JsonPatchConflict
 from .validated_config_db_connector import ValidatedConfigDBConnector
+from swsscommon.swsscommon import validate_interface_name_length, iface_name_max_length
 
 ADHOC_VALIDATION = True
+IFNAMSIZ = 16
```
Comment (qiluo-msft) [inline config/vxlan.py:9]: >16 [](http://example.com/codeflow?start=11&length=2)

Is it possible to get the value programmatically? In future, Linux may support longer ifName.

File: config/vxlan.py [hunk 2]
```diff
@@ -23,7 +25,9 @@ def add_vxlan(db, vxlan_name, src_ip):
 
     if ADHOC_VALIDATION:
```
Comment (qiluo-msft) [inline config/vxlan.py:26]: >ADHOC_VALIDATION [](http://example.com/codeflow?start=7&length=16)

The design of ADHOC_VALIDATION should be on-par with yang validation. Could you ensure this constrain is also added into sonic yang models?

File: config/main.py [hunk 3]
```diff
@@ -2400,8 +2400,9 @@ def add_portchannel(ctx, portchannel_name, min_links, fallback, fast_rate):
     db = ValidatedConfigDBConnector(ctx.obj['db'])
     if ADHOC_VALIDATION:
         if is_portchannel_name_valid(portchannel_name) != True:
-            ctx.fail("{} is invalid!, name should have prefix '{}' and suffix '{}'"
-                    .format(portchannel_name, CFG_PORTCHANNEL_PREFIX, CFG_PORTCHANNEL_NO))
+            ctx.fail("{} is invalid!, name should have prefix '{}' and suffix '{}' and its length should not \
+            exceed {} characters".format(portchannel_name, CFG_PORTCHANNEL_PREFIX, CFG_PORTCHANNEL_NO,
```
Comment (qiluo-msft) [inline config/main.py:2404]: >exceed [](http://example.com/codeflow?start=12&length=6)

The new line is so confusing. Could you reformat with some code formatter like pep8?

File: config/main.py [hunk 4]
```diff
@@ -6849,8 +6850,8 @@ def add_loopback(ctx, loopback_name):
     config_db = ValidatedConfigDBConnector(ctx.obj['db'])
     if ADHOC_VALIDATION:
         if is_loopback_name_valid(loopback_name) is False:
-            ctx.fail("{} is invalid, name should have prefix '{}' and suffix '{}' "
-                    .format(loopback_name, CFG_LOOPBACK_PREFIX, CFG_LOOPBACK_NO))
+            ctx.fail("{} is invalid, name should have prefix '{}' and suffix '{}' and should not exceed {} \
+            characters".format(loopback_name, CFG_LOOPBACK_PREFIX, CFG_LOOPBACK_NO, iface_name_max_length()))
```
Comment (qiluo-msft) [inline config/main.py:6854]: >characters [](http://example.com/codeflow?start=12&length=10)

reformat

File: config/vxlan.py (hunk 1 above)
Comment (qiluo-msft) [reply config/vxlan.py:9]: or centralize it inside sonic repos

Comment (arfeigin) [reply config/vxlan.py:9]: Thanks.
I removed it, this actually wasn't used (I forgot to delete it) - I was using the value imported from sonic-swss-common: line #29

File: config/vxlan.py (hunk 2 above)
Comment (arfeigin) [reply config/vxlan.py:25]: Added yang-models validation as well: https://github.com/sonic-net/sonic-buildimage/pull/20108

File: config/main.py (hunk 3 above)
Comment (arfeigin) [reply config/main.py]: Done

File: config/main.py (hunk 4 above)
Comment (arfeigin) [reply config/main.py]: Done

**Review by qiluo-msft — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Comment by stepanblyschak** [comment]
@qiluo-msft @prsunny Please review https://github.com/sonic-net/sonic-utilities/pull/3580

**Pull request closed without merging**